dataflowAPI/src/slicing.C

```diff
@@ -776,42 +776,45 @@ Slicer::getPredecessors(
 
     // Case 1: intra-block
     if(prev != cand.loc.rend) {
-        SliceFrame nf(cand.loc,cand.con);
-        nf.loc.rcurrent = prev;
-
-        // Slightly more complicated than the forward case; check
-        // a predicate for each active abstract region to see whether
-        // we should continue
-        bool cont = false;
-        SliceFrame::ActiveMap::const_iterator ait = cand.active.begin();
-        for( ; ait != cand.active.end(); ++ait) {
-            bool add = p.addPredecessor((*ait).first);
-            if(add)
-                nf.active.insert(*ait);
-            cont = cont || add;
+      SliceFrame *nf = NULL;
+
+      //Slightly more complicated than the forward case; check
+      //a predicate for each active abstract region to see whether
+      //we should continue
+      bool cont = false;
+      SliceFrame::ActiveMap::const_iterator ait = cand.active.begin();
+      for( ; ait != cand.active.end(); ++ait) {
+        bool add = p.addPredecessor((*ait).first);
+        if(add) {
+          if (nf == NULL) {
+            newCands.emplace_back(cand.loc, cand.con);
```

Comment:
@hainest, it's not missing. I should be clearer. Here we allocate an entry in `newCands` and grab the entry's pointer.

In contrast, previously we insert elements into `nf.active.insert` and then copy all map elements to `newCands`. With the new code, the copy process is eliminated.

Reply:
Ah, ok. I was thinking of a different type of pre-allocation. This makes sense.